onnxruntime/core/mlas/lib/mlasi.h

```diff
@@ -2062,3 +2062,51 @@ MlasReadTimeStampCounter(void)
 #endif
 #endif
 }
+
+//
+// Aligned buffer for GEMM packing, etc.
+//
+
+
+constexpr size_t ThreadedBufAlignment = 64;
+extern thread_local size_t ThreadedBufSize;
+#ifdef _MSC_VER
+extern thread_local std::unique_ptr<uint8_t, decltype(&_aligned_free)> ThreadedBufHolder;
+#else
+extern thread_local std::unique_ptr<uint8_t, decltype(&free)> ThreadedBufHolder;
+#endif
```

Comment:
If we make them local variable of MlasThreadedBufAlloc, those global objects won't be created if user inference a full-precision model

Reply:
Full-precision model should use this buffer too. We need to fix those in future PRs. And this function is inlined (copied) into multiple places. I am worried that this would create several copies of the variable, which is highly undesirable.